lcls-twincat-motion/Library/Tests/FB_MotionVirtualFrame_Test.TcPOU

```diff
@@ -0,0 +1,1135 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_MotionVirtualFrame_Test" Id="{89c3fcc0-8932-48a8-9976-db5e1ce2e9c1}" SpecialFunc="None">
+    <Declaration><![CDATA[
+FUNCTION_BLOCK FB_MotionVirtualFrame_Test EXTENDS TcUnit.FB_TestSuite
+VAR
+    nTestIDAssigner : INT;
+    nTestID : INT := 1;
+    tonTimeout : TON;
+    tonStart : TON;
+
+    stMotionStageRx : ST_MotionStage;
+    stMotionStageRy : ST_MotionStage;
+    stMotionStageRz : ST_MotionStage;
+
+    stMotionStageVx : ST_MotionStage;
+    stMotionStageVy : ST_MotionStage;
+    stMotionStageVz : ST_MotionStage;
+
+    fbMotionStageRx : FB_MotionStage;
+    fbMotionStageRy : FB_MotionStage;
+    fbMotionStageRz : FB_MotionStage;
+
+    fbMotionStageVx : FB_MotionStage;
+    fbMotionStageVy : FB_MotionStage;
+    fbMotionStageVz : FB_MotionStage;
+
+    nPositionCountsVx AT %Q* : UDINT;
+    nPositionCountsVy AT %Q* : UDINT;
+    nPositionCountsVz AT %Q* : UDINT;
+
+    fbMotorTestSuite : FB_MotorTestSuite;
+    fbMotionStageSetAndMoveHelper : FB_MotionStageSetAndMoveHelper;
+    fbMotionStageSetHelperRx : FB_MotionStageSetHelper;
+    fbMotionStageSetHelperRy : FB_MotionStageSetHelper;
+    fbMotionStageSetHelperRz : FB_MotionStageSetHelper;
+    fbMotionStageSetHelperVx : FB_MotionStageSetHelper;
+    fbMotionStageSetHelperVy : FB_MotionStageSetHelper;
+    fbMotionStageSetHelperVz : FB_MotionStageSetHelper;
+    fbMotionVirtualFrame : FB_MotionVirtualFrame;
+END_VAR
+
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[
+nTestIDAssigner := 0;
+
+tonStart(IN := TRUE, PT := T#1s);
+
+TestEnabled000045XYZMoveVxExpectNoErrorsCorrectPosVelAcc();
+
+TestEnabled000090XYZMoveVxExpectNoErrorsCorrectPosVelAcc();
+
+TestEnabled009090XYZMoveVxExpectNoErrorsCorrectPosVelAcc();
+
+TestEnabled909090XYZMoveVxExpectNoErrorsCorrectPosVelAcc();
+
+TestEnabledN22N21P4XYZMoveVxVyVzExpectNoErrorsCorrectPosVelAcc();
+
+TestEnabledN22N21P4XYZMoveVyExpectNoErrorsCorrectPosVelAcc();
```

Comment:
I don't think these quite follow your own guidelines for unit test naming, unless I'm misunderstanding.

Reply:
They are more complicated than I would like but that is due to the motion library being structured in such a way that I cannot test new functions that rely on the motion stage code without doing larger and more complicated unit tests that are difficult to describe.

Reply:
I think a future improvement to the motion library would be to break the function blocks used within an FB_MotionStage into a set of interfaces that are provided at instantiation of an FB_MotionStage. That way I could mock it out in smaller pieces and even mock an NC axis so I don't have to use simulation axes anymore.